docs/rfc/0003-collaborative-objects.md

```diff
@@ -0,0 +1,727 @@
+from other actors you just pass these changes (which, again, are just bytes) to
+automerge to add to the change log. The end result is that you load a bunch of
+binary changes and get back a JSON object.
+
+There are some subtleties around preserving user intent when modifying
+documents, but these are not too onerous.
+
+### Change Commits
+
+Given that automerge changes are a hash linked graph, we can map them to Git.
+We do so by wrapping each change in a commit. The commit points at a tree with
+the following layout
+
+```
+.
+|--change
+|--manifest.toml
+```
+
+This tree contains a single change to a collaborative object. We will go into
+more details shortly. Any direct dependencies of this change are encoded in the 
+same manner and become the parents of this commit. This allows us to
+reconstruct the automerge depdency graph. 
+
+Along with the dependencies of the commit we also need to add the commit of the
+identity which created this commit. We need this identity to validate
+signatures and by making the commit a parent we ensure that git will replicate
+it for us. 
+
+A valid change commit must have three trailers:
+
+- `X-Rad-Signature`, as for identity documents
+- `X-Rad-Author-Parent`, this is the hash of the commit which references the
+  author identity. We use this trailer to avoid following the author commit
+  reference when constructing the automerge change graph
+- `X-Rad-Schema-Parent`, this is the hash of the parent commit which contains 
+  the schema of this object. See [schema commits](#schema-commits).
+
+
+#### `manifest.toml`
+
+The manifest is a TOML file containing some metadata about the object.
+Specifically it will contain:
+
+- `id`, a UUID, generated at the time the object is created
+- `typename`, discussed above
+- `history_type`, always `"automerge"`, this is here to allow for different
+  CRDT implementations in future.
+
+
+Each object is also created with a JSON schema. The schema is represented by an
+initial `schema.json` and a series of schema migrations which extend that
+initial schema. Schema migrations will not be addressed in detail in this RFC
+but we will show their feasibility.
+
+#### `change`
+
+This is the automerge change which this commit introduces. It is a binary file
+which must contain a single change and it's dependents must be the dependents
+referenced by the parents of the commit.
+
+
+### Reconstructing Collaborative Objects
+
+Assuming we have replicated a number of collaborative objects from our tracking
+graph, we can now view the merged state of those objects. To do this we search
+through every `/rad/collaborative-objects/<typename>/<object ID>` reference for
+every remote we have and collect the change files for each object ID.
+
+At this point we have the hash linked graph of automerge changes, but we need
+to make sure that the merged document is authenticated and valid with respect
+to it's schema. To do this we start at the roots of the hash graph and walk
```

Comment:
Is the plural “roots” intentional? If I understand this correctly, there is only one root: the change that created the document. It would also help clarify to make explicit what the root is in the RFC.